packages/ts/react-crud/src/locale-formatter.ts

```diff
@@ -0,0 +1,105 @@
+import type { DatePickerDate } from '@hilla/react-components/DatePicker.js';
+
+function getFormatRegex(format: Intl.DateTimeFormat) {
+  const sampleDate = new Date(1234, 5 - 1, 6);
+  const formattedSample = format.format(sampleDate);
+  const pattern = formattedSample
+    .replace('1234', '(?<year>\\d+)')
+    .replace('05', '(?<month>\\d+)')
+    .replace('5', '(?<month>\\d+)')
+    .replace('06', '(?<day>\\d+)')
+    .replace('6', '(?<day>\\d+)');
+
+  return new RegExp(pattern, 'u');
+}
+
+export interface DateObject {
+  year: number;
+  month: number;
+  day: number;
+}
+
+function tryFormatDate(formatter: Intl.DateTimeFormat, value?: string): string {
+  try {
+    return value ? formatter.format(new Date(value)) : '';
+  } catch (e) {
+    return '';
+  }
+}
+
+export class LocaleFormatter {
```

Comment:
`LocaleFormatter` seems like a worse name, since we're formattings dates, not locales.

Reply:
I have added a `NumberFormat` here as well because it depends on locale, so the `Date` part sounds misleading. However, I'm open to any naming suggestions 🙂

Reply:
Ah ok, I missed the number formatting. Let's keep it as is then.